packages/rps/src/redux/message-service/saga.ts

```diff
@@ -267,29 +277,58 @@ function* receiveChallengePositionFromWalletSaga() {
   }
 }
 
+function relayCommitmentToWallet(c: Commitment, s: string) {
+    Wallet.messageWallet(WALLET_IFRAME_ID, c, s);
+}
+
+function* receiveCommitmentSaga(message: Message) {
+  const { data, signature, userName } = message;
+  const validMessage = yield validateMessage(data, signature);
+  if (!validMessage) {
+    // TODO: Handle this
+  }
+
+  if (data.turnNum === 0) {
+    yield put(gameActions.initialCommitmentReceived(fromCoreCommitment(data), userName ? userName : 'Opponent'));
+  } else {
+    yield put(gameActions.commitmentReceived(fromCoreCommitment(data)));
+  }
+}
+
+async function postData(data = {}) {
+  const response = await fetch(`${process.env.REACT_APP_BOT_URL}/api/v1/rps_channels`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data),
+  });
+  return await response.json(); // parses response to JSON
+}
 
-function* validateMessage(commitment: RPSCommitment, signature) {
+function* validateMessage(commitment: Commitment, signature) {
   try {
-    return yield Wallet.validateCommitmentSignature(WALLET_IFRAME_ID, asCoreCommitment(commitment), signature);
+    return yield Wallet.validateCommitmentSignature(WALLET_IFRAME_ID, commitment, signature);
   } catch (err) {
     if (err.reason === 'WalletBusy') {
+      // Why do you create this event channel? What does 'WalletBusy' really mean?
```

Comment:
The `wallet` can refuse to sign or validate something if a challenge is ongoing (and possibly some other scenarios?) so  application need to be able to handle the case where the wallet is busy and try again later.
We could rename `WalletBusy` to `ChallengeOngoing` to make this clearer. (maybe add a message as well)

Reply:
👍